packages/@aws-cdk/aws-cloudwatch/lib/alarm.ts

```diff
@@ -159,6 +159,21 @@ export class Alarm extends Resource implements IAlarm {
 
     const comparisonOperator = props.comparisonOperator || ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD;
 
+    // Render metric, process potential overrides from the alarm
+    // (It would be preferable if the statistic etc. was worked into the metric,
+    // but hey we're allowing overrides...)
+    const metricProps: Writeable<Partial<CfnAlarmProps>> = this.renderMetric(props.metric);
+    if (props.period) {
+      metricProps.period = props.period.toSeconds();
+    }
+    if (props.statistic) {
+      // Will overwrite both fields if present
+      Object.assign(metricProps, {
+        statistic: renderIfSimpleStatistic(props.statistic),
+        extendedStatistic: renderIfExtendedStatistic(props.statistic),
+      });
+    }
```

Comment:
These `if`s are technically not required - 

```suggestion
    metricProps.period = props.period?.toSeconds();
    Object.assign(metricProps, {
      statistic: renderIfSimpleStatistic(props.statistic),
      extendedStatistic: renderIfExtendedStatistic(props.statistic),
    });
```

Reply:
```
    metricProps.period = props.period?.toSeconds();
```

It's not the same. This will unconditionally overwrite `metricProps.period` with `undefined`, whereas the other one would overwrite `metricProps.period` only if there was a value to set.

Same for the other one.

Reply:
Gah.. you're right!